### PR TITLE
BCI java: memory allocation and garbage colleciton

### PR DESCRIPTION
**Pull request by ilausuch**
http://progress.opensuse.org/issues/110689
VR: http://marvin5.arch.suse.cz/tests/756

### CODE REVIEW

File: tests/test_openjdk.py [hunk 1]
```diff
@@ -67,23 +67,42 @@ def test_jdk_version(container, java_version):
...
+            "memory",
+            ["Iteration: (2)", "OutOfMemoryError"],
+            [1],
+            "-Xmx10M",
+        ),
+        (
+            "memory",
+            ["Iteration: (45)", "OutOfMemoryError"],
+            [1],
+            "-Xmx100M",
```
Comment (cfconrad) [inline tests/test_openjdk.py:95]: Funny Java, that we need twice the memory as we actually allocate? Or is it copied when we call
```
v.add(b);
```
??

File: tests/trainers/java/memory.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+  public static void main(String[] args)
+  {
+    int iter = 0;
+    Vector v = new Vector();
+    while (true)
+    {
+      byte b[] = new byte[1048576];
+      v.add(b);
+      Runtime rt = Runtime.getRuntime();
```
[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/trainers/java/memory.java:19]: yes, this is a left over

File: tests/test_openjdk.py (hunk 1 above)
Comment (ilausuch) [reply tests/test_openjdk.py:95]: The idea of this test is to check if parameter Xmx works effectively, so first I allocate 10Mb and then 100Mb seeing witch iteration we have. Maybe I two tests for the same is not necessary

File: tests/trainers/java/memory.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/trainers/java/memory.java]: updated

File: tests/test_openjdk.py (hunk 1 above)
Comment (ilausuch) [reply tests/test_openjdk.py]: I removed the second one

Comment (cfconrad) [reply tests/test_openjdk.py]: ok, this wasn't my intention, sorry for the confusion.

**Review by cfconrad — APPROVED** [review]
LGTM

**Review by dcermak — CHANGES_REQUESTED** [review]
Please rebase on main

File: tests/test_openjdk.py [hunk 3]
```diff
@@ -67,23 +67,36 @@ def test_jdk_version(container, java_version):
...
+            "-Xmx10M",
+        ),
+        ("garbage_collector", [], [0], ""),
     ],
 )
 def test_jdk_extended(
     container_per_test,
     test_to_run: str,
     expected_strings: List[str],
+    expected_exit_status: int,
```
[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/test_openjdk.py:98]: ohh good catch!

**Comment by dcermak** [comment]
@ilausuch the branch is still stale

**Comment by ilausuch** [comment]
Ok, thanks! rebased now

**Review by dcermak — APPROVED** [review]

**Pull request merged**